nginx/build.go

```diff
@@ -0,0 +1,153 @@
+	Install(dependency postal.Dependency, cnbPath, layerPath string) error
+}
+
+//go:generate faux --interface ProfileDWriter --output fakes/profiled_writer.go
+type ProfileDWriter interface {
+	Write(layerDir, scriptName, scriptContents string) error
+}
+
+//go:generate faux --interface Calculator --output fakes/calculator.go
+type Calculator interface {
+	Sum(path string) (string, error)
+}
+
+func Build(entryResolver EntryResolver, dependencyService DependencyService, profileDWriter ProfileDWriter, calculator Calculator, logger LogEmitter, clock Clock) packit.BuildFunc {
+	return func(context packit.BuildContext) (packit.BuildResult, error) {
+		logger.Title(context.BuildpackInfo)
+
+		logger.Process("Resolving Nginx Server version")
+		logger.Candidates(context.Plan.Entries)
+
+		entry := entryResolver.Resolve(context.Plan.Entries)
+
+		dependency, err := dependencyService.Resolve(filepath.Join(context.CNBPath, "buildpack.toml"), entry.Name, entry.Version, context.Stack)
+		if err != nil {
+			panic(err)
+		}
+
+		logger.SelectedDependency(entry, dependency.Version)
+
+		err = os.MkdirAll(filepath.Join(context.WorkingDir, "logs"), os.ModePerm)
+		if err != nil {
+			panic(err)
+		}
+
+		nginxLayer, err := context.Layers.Get(NGINX, packit.LaunchLayer)
+		if err != nil {
+			panic(err)
+		}
+
+		nginxConfPath := filepath.Join(context.WorkingDir, ConfFile)
+		defaultStartProcesses := []packit.Process{
+			{
+				Type:    "web",
+				Command: fmt.Sprintf(`nginx -p $PWD -c "%s"`, nginxConfPath),
+			},
+		}
+
+		currConfigureBinSHA256, err := calculator.Sum(filepath.Join(context.CNBPath, "bin", "configure"))
+		if err != nil {
+			panic(err)
+		}
+
+		run := shouldInstall(nginxLayer.Metadata, currConfigureBinSHA256, dependency.SHA256)
+		if !run {
+			return packit.BuildResult{
+				Plan: context.Plan,
+				Layers: []packit.Layer{
+					nginxLayer,
+				},
+				Processes: defaultStartProcesses,
+			}, nil
+		}
+
+		logger.Break()
+		logger.Process("Executing build process")
+
+		err = nginxLayer.Reset()
+		if err != nil {
+			panic(err)
+		}
+
+		err = CopyBinFile(filepath.Join(nginxLayer.Path, "bin", "configure"), filepath.Join(context.CNBPath, "bin", "configure"))
+		if err != nil {
+			panic(err)
+		}
+
+		logger.Subprocess("Installing Nginx Server %s", dependency.Version)
+		then := clock.Now()
+		err = dependencyService.Install(dependency, context.CNBPath, nginxLayer.Path)
+		if err != nil {
+			panic(err)
+		}
+
+		logger.Action("Completed in %s", time.Since(then).Round(time.Millisecond))
+		logger.Break()
+
+		logger.Process("Configuring environment")
+		nginxLayer.SharedEnv.Append("PATH", filepath.Join(nginxLayer.Path, "sbin"), ":")
+		logger.Environment(nginxLayer.SharedEnv)
+
+		profileDWriter.Write(
+			nginxLayer.Path,
+			"configure.sh",
+			fmt.Sprintf(`configure "%s" "%s" "%s"`,
+				nginxConfPath,
+				filepath.Join(context.WorkingDir, "modules"),
+				filepath.Join(nginxLayer.Path, "modules"),
+			),
+		)
```

Comment:
I can't see that we log that a profile.d script is being added to the container. We have done this in other buildpacks like [node-engine](https://github.com/paketo-buildpacks/node-engine/blob/b3c16b06a08cdca59e4d9470ca969398b4c985a0/node/log_emitter.go#L56-L58) when they have these scripts. It might make some sense to do something similar here.

Reply:
Done